test/cadmium_dynamic_LI_test.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2013-2016, Damian Vicino
+ * Carleton University, Universite de Nice-Sophia Antipolis
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include <boost/test/unit_test.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/data/monomorphic.hpp>
+
+#include "test_helpers.hpp"
+#include "../src/dynamic/LI_generator.cpp"
+#include "../../DEVSDiagrammer/model_json_exporter/include/dynamic_json_exporter.hpp"
+
+namespace pt = boost::property_tree;
+namespace bdata = boost::unit_test::data;
+
+BOOST_AUTO_TEST_SUITE( cadmium_dynamic_LI_test_suite)
```

Comment:
My idea was to output the model to a json file, have a directory having a list of "expected" json files, and a script that checks all generated files are "equal" to those expected.
Part of the code below can be used for a small command line binary comparing 2 files and telling they are "json" equal enough, or maybe we can jut relay on python or other language already supporting the comparison. 
However, if that below works, it already seems why more flexible and simple than what we had before.

Reply:
Making the tests in python starting with the json files sound better, the way you have to access the json using the property_tree isn't very good.

On the other hand, i think checking properties on the json structure is better than checking equallity. Specially thinking about having the same tests to run for multiple simulated versions; at least for the two cadmium versions, the json is no exactly the same (names of models, we actually read the lists in the json as sets, etc).